test/langtools/tools/javac/reflect/SwitchExpressionTest2.java

```diff
@@ -0,0 +1,633 @@
+                %1 : Var<java.lang.String> = var %0 @"r";
+                %2 : java.lang.String = var.load %1;
+                %3 : java.lang.String = java.switch.expression %2
+                    (%4 : java.lang.String)boolean -> {
+                        %5 : java.lang.String = constant @"FOO";
+                        %6 : boolean = invoke %4 %5 @"java.util.Objects::equals(java.lang.Object, java.lang.Object)boolean";
+                        yield %6;
+                    }
+                    ()java.lang.String -> {
+                        %7 : java.lang.String = constant @"BAR";
+                        yield %7;
+                    }
+                    (%8 : java.lang.String)boolean -> {
+                        %9 : java.lang.String = constant @"BAR";
+                        %10 : boolean = invoke %8 %9 @"java.util.Objects::equals(java.lang.Object, java.lang.Object)boolean";
+                        yield %10;
+                    }
+                    ()java.lang.String -> {
+                        %11 : java.lang.String = constant @"BAZ";
+                        yield %11;
+                    }
+                    (%12 : java.lang.String)boolean -> {
+                        %13 : java.lang.String = constant @"BAZ";
+                        %14 : boolean = invoke %12 %13 @"java.util.Objects::equals(java.lang.Object, java.lang.Object)boolean";
+                        yield %14;
+                    }
+                    ()java.lang.String -> {
+                        %15 : java.lang.String = constant @"FOO";
+                        yield %15;
+                    }
+                    ()void -> {
+                        yield;
+                    }
+                    ()java.lang.String -> {
+                        %16 : java.lang.String = constant @"";
+                        yield %16;
+                    };
+                return %3;
+            };
+            """)
+    @CodeReflection
+    public static String caseConstantRuleExpression(String r) {
+        return switch (r) {
+            case "FOO" -> "BAR";
+            case "BAR" -> "BAZ";
+            case "BAZ" -> "FOO";
+            default -> "";
+        };
+    }
+
+    @IR("""
+            func @"caseConstantRuleBlock" (%0 : java.lang.String)java.lang.String -> {
+                %1 : Var<java.lang.String> = var %0 @"r";
+                %2 : java.lang.String = var.load %1;
+                %3 : java.lang.String = java.switch.expression %2
+                    (%4 : java.lang.String)boolean -> {
+                        %5 : java.lang.String = constant @"FOO";
+                        %6 : boolean = invoke %4 %5 @"java.util.Objects::equals(java.lang.Object, java.lang.Object)boolean";
+                        yield %6;
+                    }
+                    ()java.lang.String -> {
+                        %7 : java.lang.String = constant @"BAR";
+                        java.yield %7;
```

Comment:
unrelated: but I found the distinction between `yield` and `java.yield` a bit confusing...

Reply:
Indeed it is confusing. The core operation `yield` means to yield a value from a body, where as `java.yield` models Java's `yield` statement. Perhaps rename the former to `body.yield`?